### PR TITLE
Added new docs for plugin feature flags

### PR DESCRIPTION
**Pull request by blueboxes**
Signed-off-by: John Kilmister <blueboxes@users.noreply.github.com>

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by blueboxes** [comment]
I am fixing the format however feedback on content welcome in the meanwhile

File: mkdocs.yml [hunk 1]
```diff
@@ -127,10 +127,10 @@ nav:
       - Create a Backstage Plugin: 'plugins/create-a-plugin.md'
       - Plugin Development: 'plugins/plugin-development.md'
       - Structure of a plugin: 'plugins/structure-of-a-plugin.md'
-      - Plugin Development: 'plugins/plugin-development.md'
-      - Integrate into the Software Catalog: 'plugins/integrating-plugin-into-software-catalog.md'
+      - Integrate into the Software Cagittalog: 'plugins/integrating-plugin-into-software-catalog.md'
```
Comment (awanlin) [inline mkdocs.yml:130]: Small typo here

File: docs/plugins/feature-flags.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+---
+id: feature-flags
+title: Feature Flags
+description: Details the process of defining setting and reading a plugin feature flag.
+---
+
+Backstage offers the ability to define feature flags inside a plugin. This allows you to restrict parts of your plugin to those individual users who have toggled the feature flag to on.
+
+This page describes the process of defining setting and reading a plugin feature flag.
```
Comment (awanlin) [inline docs/plugins/feature-flags.md:9]: I'm wondering if we could add another line to this: "If you are looking for using Feature flags with software templates that can be found under [Writing Templates](https://backstage.io/docs/features/software-templates/writing-templates#remove-sections-or-fields-based-on-feature-flags)"

File: docs/plugins/feature-flags.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+The easiest way to control content based on the state of a feature flag is to use the [FeatureFlagged](https://backstage.io/docs/reference/core-app-api.featureflagged) component.
+
+```ts
+import { FeatureFlagged } from '@backstage/core-app-api'
+
+...
+
+<FeatureFlagged with="show-example-feature">
+    <SampleComponent />
```
Comment (awanlin) [inline docs/plugins/feature-flags.md:48]: Could this be `<NewFeatureComponent />` and then on for the "without" be `<PreviousFeatureComponent />`, just to better illustrate the difference?

**Review by awanlin — COMMENTED** [review]
Awesome work @blueboxes, really appreciate you taking the time to write this up as will help many people 👍 🚀 

Just left a few small comments

File: mkdocs.yml (hunk 1 above)
Comment (blueboxes) [reply mkdocs.yml:130]: Ha you can see I typed git and cursor was in the wrong place and it edited the file 🤦‍♂️ good catch

File: docs/plugins/feature-flags.md (hunk 3 above)
Comment (blueboxes) [reply docs/plugins/feature-flags.md:48]: Good idea

**Comment by blueboxes** [comment]
I have replaced this PR with https://github.com/backstage/backstage/pull/13623 for cleaner history

**Pull request closed without merging**